qcodes/instrument_drivers/QDev/QDac.py

```diff
@@ -0,0 +1,484 @@
+            rampchans = ', '.join([str(c[0]) for c in self._slopes])
+            raise ValueError('Can not assign finite slope to more than ' +
+                             "8 channels. Assign 'Inf' to at least one of " +
+                             'the following channels: {}'.format(rampchans))
+
+        self._slopes.append([chan, slope])
+        return
+
+    def _getslope(self, chan):
+        """
+        get_cmd of the chXX_slope parameter
+        """
+        if chan in [sl[0] for sl in self._slopes]:
+            slope = [sl[1] for sl in self._slopes if sl[0]==chan][0]
+            return slope
+        else:
+            return 'Inf'
+
+    def slopes(self):
+        """
+        unit the slopes assigned to each channel
+        """
+        for (chan, slope) in self._slopes:
+            print('Channel {}, slope: {} (V/s)'.format(chan, slope))
+
+    def _rampvoltage(self, chan, fg, setvoltage, ramptime):
+        """
+        Smoothly ramp the voltage of a channel by the means of a function
+        generator. Helper function used by _set_voltage.
+
+        Args:
+            chan (int): The channel number (counting from 1)
+            fg (int): The function generator (counting from 1)
+            setvoltage (float): The voltage to ramp to
+            ramptime (float): The ramp time in seconds.
+        """
+
+        v_start = self.parameters['ch{:02}_v'.format(chan)].get_latest()
+
+        offset = v_start
+        amplitude = setvoltage-v_start
+        if self.parameters['ch{:02}_vrange'.format(chan)].get_latest() == 1:
+            offset *= 10
+            amplitude *= 10
+
+        chanmssg = 'wav {} {} {} {}'.format(chan, fg,
+                                            amplitude,
+                                            offset)
+
+        typedict = {'SINE': 1, 'SQUARE': 2, 'RAMP': 3}
+
+        typeval = typedict['RAMP']
+        dutyval = 100
+        # s -> ms
+        periodval = ramptime*1e3
+        repval = 1
+        funmssg = 'fun {} {} {} {} {}'.format(fg,
+                                              typeval, periodval,
+                                              dutyval, repval)
+        self.write(chanmssg)
+        self.write(funmssg)
+        self.parameters['ch{:02}_v'.format(chan)]._save_val(setvoltage)
+
+    def write(self, cmd):
+        """
+        QDac always returns something even from set commands, even when
+        verbose mode is off, so we'll override write to take this out
+        if you want to use this response, we put it in self._write_response
+        (but only for the very last write call)
+
+        Note that this procedure makes it very cumbersome to handle the returned
+        messages from concatenated commands, e.g. 'wav 1 1 1 0;fun 2 1 100 1 1'
+        Please don't use concatenated commands
+        """
+        nr_bytes_written, ret_code = self.visa_handle.write(cmd)
+        self.check_error(ret_code)
+        self._write_response = self.visa_handle.read()
```

Comment:
Should perhaps define _write_response as an empty string in `__init__` so it's always defined